### PR TITLE
(feature) implement reserve balance comparison for valid merkato init

### PR DESCRIPTION
**Pull request by mark-antony1**

### CODE REVIEW

File: merkato/utils/__init__.py [hunk 1]
```diff
@@ -71,3 +71,29 @@ def generate_complete_merkato_configs(merkato_tuples):
...
+            allocated_pair_balances['coin'] += merkato['ask_reserved_balance']
+    return allocated_pair_balances
+
+def check_reserve_balances(total_balances, allocated_balances, coin_reserve, base_reserve):
+    remaining_balances = {
+        'base': total_balances['base']['amount'] - allocated_balances['base'],
+        'coin': total_balances['coin']['amount'] - allocated_balances['coin']
+    }
+    if remaining_balances['base'] < base_reserve:
+        raise ValueError('Cannot create merkato, the suggested base reserve will exceed the amount of the base asset on the exchange.')
```
Comment (needmoney90) [inline merkato/utils/__init__.py:96]: Why is the 'check reserve balances' function aware of the 'create merkatos' function?

Comment (needmoney90) [reply merkato/utils/__init__.py:96]: in terms of it's error message

Comment (mark-antony1) [reply merkato/utils/__init__.py:96]: the only time its called is in the init of the merkato. it fails and stops merkato creation from happening

Comment (needmoney90) [reply merkato/utils/__init__.py:96]: The function doesn't have a return value, or modify any variables. Can you restructure it such that it returns a truthy value, and the parent function raises the error in its own context? I'm not comfortable with this util function being context aware of the function that uses it.

**Review by needmoney90 — APPROVED** [review]

**Pull request merged**